### PR TITLE
doc/voice (and VoiceGroup removed)

### PR DESCRIPTION
**Pull request by rvilarl**
- VoiceGroup Removed 
- Voice Documented

I have taken the following approach (to be agreed):
- Exclusion of protected and private members in the ouput from typedoc.
- Usage of imperative rather that third person in comments.
- Protection of methods only used within the class.
- Simple Header: product, copyright and license (the license statement on each file help the automatic sw clearance tools) 

@0xfe @AaronDavidNewman @mscuthbert @ronyeh 
I have tried to address the concerns in #1004. Once we agree with this PR we can address the next ones in a similar well.

### CODE REVIEW

File: src/voice.ts [hunk 1]
```diff
@@ -28,12 +22,15 @@ export enum VoiceMode {
   FULL = 3,
 }
 
+/**
+ * Voice class is mainly a container
+ * object to group `Tickables` for formatting.
+ */
```
Comment (ronyeh) [inline src/voice.ts:28]: How about:

\`Voice\` is mainly a container object to group \`Tickables\` for formatting.

File: src/voice.ts [hunk 2]
```diff
@@ -44,12 +41,14 @@ export class Voice extends Element {
...
-  // Modes allow the addition of ticks in three different ways:
-  //
-  // STRICT: This is the default. Ticks must fill the voice.
-  // SOFT:   Ticks can be added without restrictions.
-  // FULL:   Ticks do not need to fill the voice, but can't exceed the maximum
-  //         tick length.
+  /**
+   * Modes allow the addition of ticks in three different ways:
+   *
+   * STRICT: This is the default. Ticks must fill the voice.<br>
```
Comment (ronyeh) [inline src/voice.ts:47]: Is the \<br\> tag the only way to force a line break? Is the alternative to have an extra empty line between the STRICT line and SOFT line?

Personally I would prefer having as few HTML tags in the comments as possible, but I could live with the br tags if there is no good alternative.

What about putting the three different ways in a bulleted list?

```
Modes allow the addition of ticks in three different ways:

- STRICT: This is the default. Ticks must fill the voice.
- SOFT:   Ticks can be added without restrictions.
- FULL:   Ticks do not need to fill the voice, but can't exceed the maximum tick length.
```

File: src/voice.ts [hunk 3]
```diff
@@ -99,59 +98,63 @@ export class Voice extends Element {
...
   }
 
-  // Get the total ticks used in the voice by all the tickables
+  /** Get the total ticks used in the voice by all the tickables */
   getTicksUsed(): Fraction {
     return this.ticksUsed;
   }
 
-  // Get the largest width of all the tickables
+  /** Get the largest width of all the tickables */
```
Comment (ronyeh) [inline src/voice.ts:111]: Super duper minor, but I would prefer that sentence comments end with a period.

File: src/voice.ts [hunk 4]
```diff
@@ -205,26 +192,30 @@ export class Voice extends Element {
...
     this.options.softmaxFactor = factor;
     return this;
   }
 
-  // Calculate the sum of the exponents of all the ticks in this voice to use as the denominator
-  // of softmax.
-  reCalculateExpTicksUsed(): number {
+  /**
+   * Calculate the sum of the exponents of all the ticks in this voice to use as the denominator
+   * of softmax.
```
Comment (ronyeh) [inline src/voice.ts:209]: Can this all fit on one line? :-)

Overall, this style of commenting looks good!

Over time, I imagine we will add the `@param ` as needed, to explain any params that might not be obvious.

Comment (rvilarl) [reply src/voice.ts]: @param is already used with `mode`

File: src/voice.ts (hunk 3 above)
Comment (rvilarl) [reply src/voice.ts]: done

File: src/voice.ts (hunk 2 above)
Comment (rvilarl) [reply src/voice.ts]: Done

File: src/voice.ts (hunk 1 above)
Comment (rvilarl) [reply src/voice.ts:27]: Done

**Comment by rvilarl** [comment]
@ronyeh Ready

**Comment by rvilarl** [comment]
@0xfe I changed all the comments from @ronyeh, anything else from yourself?

**Comment by 0xfe** [comment]
This looks good to me.

**Pull request merged**